File: src/main/java/com/canyoncompanion/canyon_api/service/RefreshTokenService.java
package com.canyoncompanion.canyon_api.service;


import com.canyoncompanion.canyon_api.model.entities.RefreshTokenEntity;
import org.springframework.security.core.userdetails.UserDetails;
import javax.annotation.Nullable;

public interface RefreshTokenService {

    @Nullable
    RefreshTokenEntity findByToken(@Nullable String token);

    @Nullable
    String generateLoginToken(@Nullable UserDetails userDetails);

    @Nullable
    String rotateToken(@Nullable RefreshTokenEntity oldToken);
}

File: src/main/java/com/canyoncompanion/canyon_api/service/RefreshTokenService.kt
package com.canyoncompanion.canyon_api.service

import com.canyoncompanion.canyon_api.model.entities.RefreshTokenEntity
import org.springframework.security.core.userdetails.UserDetails


interface RefreshTokenService {
    fun findByToken(token: String?): RefreshTokenEntity?

    fun createOrRefreshToken(userDetails: UserDetails?): String?

    fun rotateToken(oldToken: RefreshTokenEntity?): String?
}